forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+        )
+        # create spline basis for obs period and forecast period
+        last = X[-1]
+        X_future = np.hstack(
+            (X, np.arange(last + 1, last + 1 + forecast_days))
+        )
+        sbm = spline_basis(X_future)
+        # get posterior samples
+        mcmc.run(rng_key=jr.key(random_seed), basis_matrix=sbm[: len(X)], y=y)
+        posterior_samples = mcmc.get_samples()
+        # get prior predictive
+        prior_pred = npro.infer.Predictive(model, num_samples=num_samples)(
+            rng_key=jr.key(random_seed), basis_matrix=sbm[: len(X)]
+        )
+        # get posterior predictive forecast
+        posterior_pred_for = npro.infer.Predictive(
+            model, posterior_samples=posterior_samples
+        )(rng_key=jr.key(random_seed), basis_matrix=sbm)
+        # create initial inference data object(s)
+        idata = az.from_numpyro(
+            posterior=mcmc,
+            posterior_predictive=posterior_pred_for,
+            prior=prior_pred,
+        )
+        # save idata object(s)
+        idata.to_netcdf(file_save_path)
+        # plot forecast (if desired) from idata light
+        if show_plot:
+            x_data = idata.posterior_predictive["obs_dim_0"]
+            y_data = idata.posterior_predictive["obs"]
+            fig, axes = plt.subplots(1, 1, figsize=(8, 6))
+            az.plot_hdi(
+                x_data,
+                y_data,
+                hdi_prob=0.95,
+                color="skyblue",
+                smooth=False,
+                fill_kwargs={"alpha": 0.2, "label": "95% Credible"},
+                ax=axes,
+            )
+            az.plot_hdi(
+                x_data,
+                y_data,
+                hdi_prob=0.75,
+                color="skyblue",
+                smooth=False,
+                fill_kwargs={"alpha": 0.4, "label": "75% Credible"},
+                ax=axes,
+            )
+            az.plot_hdi(
+                x_data,
+                y_data,
+                hdi_prob=0.5,
+                color="C0",
+                smooth=False,
+                fill_kwargs={"alpha": 0.6, "label": "50% Credible"},
+                ax=axes,
+            )
+            axes.plot(
+                X,
+                y,
+                marker="o",
+                color="black",
+                linewidth=1.0,
+                markersize=3.0,
+                label="Observed",
+            )
+            median_ts = y_data.median(dim=["chain", "draw"])
+            axes.plot(x_data, median_ts, color="blue", label="Median")
+            axes.legend()
+            axes.axvline(last, color="black", linestyle="--")
+            axes.set_title(
+                f"Hospital Admissions ({state}, {start_date}-{end_date})",
+                fontsize=15,
+            )
+            axes.set_xlabel("Time", fontsize=20)
+            axes.set_ylabel("Hospital Admissions", fontsize=20)
+            plt.show()
+
+
+def read_example_flusight_submission(
```

Comment:
Name could be better. It doesn't just read the example submission, it saves it. Maybe `create_example_flusight_submission`?

Reply:
I think I will go for `get_and_save_flusight_submission`.

Reply:
Renamed `read_example_flusight_submission` &rarr; `get_and_save_flusight_submission`.